R/SotkanetInteractive.R

```diff
@@ -0,0 +1,195 @@
+           title = "Would you like to use default arguments or manually select them?") +1,
+      return(invisible()),
+      FALSE,
+      TRUE
+    )
+
+    if (manual_selection) {
+      years_selection_begin <- readline(prompt = "Enter the beginning year for the data: ")
+      years_selection_end <- readline(prompt = "Enter the ending year for the data: ")
+
+      years_selection_begin <- as.integer(years_selection_begin)
+      years_selection_end <- as.integer(years_selection_end)
+
+      years <- years_selection_begin:years_selection_end
+
+      gender_selection <- switch(
+        menu(choices = c("Male", "Female", "Male & Female", "Total", "All"),
+             title = "Which genders do you want for the data?") + 1,
+        return(invisible()),
+        c("male"),
+        c("female"),
+        c("male", "female"),
+        c("total"),
+        c("male", "female", "total")
+      )
+
+      region_selection <- readline(prompt = "Enter the regions (empty for defulat): ")
+
+      region.category_selection <- readline(prompt = "Enter the region.category (empty for default): ")
+
+      if (region_selection == ""){
+        region_selection <- NULL
+      }
+
+      if (region.category_selection == ""){
+        region.category_selection <- NULL
+      }
+
+
+      sotkanet_data <- GetDataSotkanet(indicators = search_id,
+                                       years = years,
+                                       genders = gender_selection,
+                                       regions = region_selection,
+                                       region.category = region.category_selection)
+
+
+
+    } else if (!manual_selection){
+      sotkanet_data <- GetDataSotkanet(indicators = search_id)
+
+    }
+
+  }
+
+  tempfile_for_sink <- tempfile()
+
+  print_citation <- switch(
+    menu(choices = c("Yes", "No"),
+         title = "Print dataset citation?") + 1,
+    return(invisible()),
+    TRUE,
+    FALSE
+  )
+
+  if (print_citation){
+    citation <- SotkanetCite(id = search_id,
+                             lang = lang_selection)
+    capture.output(cat("#### DATASET CITATION: \n\n"),
+                   file = tempfile_for_sink, append = TRUE)
+    capture.output(print(citation),
+                   file = tempfile_for_sink, append = TRUE)
+    capture.output(cat("\n"),
+                   file = tempfile_for_sink, append = TRUE)
+  }
+
+  print_code <- switch(
+    menu(choices = c("Yes", "No"),
+         title = "Printe code for downloading dataset?") + 1,
+    return(invisible()),
+    TRUE,
+    FALSE
+  )
+
+  if (print_code == TRUE && manual_selection == TRUE){
+    capture.output(cat("#### DOWNLOAD PARAMETERS: \n\n"),
+                   file = tempfile_for_sink, append = TRUE)
+    capture.output(print(
+      paste0("GetDataSotkanet(indicators = ", search_id,
+             ", years = ", years[1], ":", years[length(years)],
+             ", genders = ", gender_selection, ", regions = ",
```

Comment:
The function works nicely but there are still some hiccups:

1. If I input region.category (and probably regions) without quotes, the function seems to accept this as valid but it still prints the download code without double quotes:

```
Enter the regions (empty for default): 
Enter the region.category (empty for default): POHJOISMAAT
[...]
#### DOWNLOAD PARAMETERS: 

[1] "get_sotkanet(indicators = 10027, years = 2000:2005, genders = c('male', 'female'), regions = NULL, region.category = POHJOISMAAT, lang = 'fi')"
```

This results in R searching for a variable called POHJOISMAAT, not string "POHJOISMAAT":

```
get_sotkanet(indicators = 10027, years = 2000:2005, genders = c('male', 'female'), regions = NULL, region.category = POHJOISMAAT, lang = 'fi')
Error: object 'POHJOISMAAT' not found
```

So I guess it's more about fine-tuning the example download string.

2. I get this weird tooltip when I input data manually:
<img width="545" alt="tooltip" src="https://github.com/rOpenGov/sotkanet/assets/69813611/a8218654-4ea0-4411-9e72-a0d85b4a3da4">

I wonder what causes it? It's obviously useless / misleading. Maybe it's from RStudio and not R?

Reply:
Then there's the hassle with selecting regions:

```
Enter the regions (empty for default): c("Suomi", "Ruotsi")
Enter the region.category (empty for default): 
Input for regions not found from dataset: c("Suomi", "Ruotsi") 
 Please check your parameter input for validity and correctness.
```

In code for downloading data this input get mangled by extra escape-characters:

```
"get_sotkanet(indicators = 10027, years = 2000:2005, genders = c('male', 'female'), regions = c(\"Suomi\", \"Ruotsi\"), region.category = NULL, lang = 'fi')"
``` 

From the API documentation we see that we should be using region codes anyway for selecting regions, not names:

```
region
^\d+$
Alueen tunnus, jota käytetään SOTKAnetin ja sen APIn kutsuissa
```

For example Sweden is 1044 and Finland is 1045. It might be a good idea either to give a hint to end user on how to input multiple values: wrapped in c(), separated by comma...?

```
Enter the regions (empty for default): 1044, 1045
Enter the region.category (empty for default): 
Input for regions not found from dataset: 1044, 1045 
 Please check your parameter input for validity and correctness.
```

```
#### DOWNLOAD PARAMETERS: 

[1] "get_sotkanet(indicators = 10027, years = 2000:2005, genders = c('male', 'female'), regions = 1044, 1045, region.category = NULL, lang = 'fi')"
```

Maybe the most ideal solution would be to read all available options for each datasets from the metadata and print them using the language that the user chose in the first step. Then the user could select one or more regions, without having to care about Sotkanet region codes.

Reply:
One way would also be to remove regions and region.category selection since the filtering is done locally anyway